src/commonMain/kotlin/com/github/quillraven/fleks/system.kt

```diff
@@ -28,15 +30,26 @@ data class Fixed(val step: Float) : Interval
  * @param interval the [interval][Interval] in which the system gets updated. Default is [EachFrame].
  * @param enabled defines if the system gets updated when the [world][World] gets updated. Default is true.
  */
-abstract class IntervalSystem(
-    val interval: Interval = EachFrame,
+abstract class IntervalSystem<T>(
+    val interval: Interval<T>? = null,
     enabled: Boolean = true,
     /**
      * Returns the [world][World] to which this system belongs.
      */
-    val world: World = World.CURRENT_WORLD ?: throw FleksWrongConfigurationUsageException()
+    val world: GenericWorld = World.CURRENT_WORLD ?: throw FleksWrongConfigurationUsageException()
 ) : EntityComponentContext(world.componentService) {
 
+    private lateinit var _clock: Clock<T>
```

Comment:
try to avoid private because there are always people out there who want to overrule something and private just makes it very hard/impossible for them ;)

Reply:
Since onUpdate() is open, is that really necessary? In the base implementation, the clock should not be mutable, and Clock is already an interface that can be extended and set at world creation.